lib/fluent/plugin/in_mysql_replicator.rb

```diff
@@ -58,27 +58,31 @@ def run
     def poll
       table_hash = Hash.new
       ids = Array.new
+      con = get_connection()
+      prepared_con = get_connection()
       loop do
         rows_count = 0
         start_time = Time.now
         previous_ids = ids
         current_ids = Array.new
-        prepared_con = get_connection()
         if !@prepared_query.nil?
           @prepared_query.split(/;/).each do |query|
             prepared_con.query(query)
           end
         end
-        query(@query).each do |row|
+        rows, con = query(@query, con)
+        rows.each do |row|
           current_ids << row[@primary_key]
           current_hash = Digest::SHA1.hexdigest(row.flatten.join)
           row.each {|k, v| row[k] = v.to_s if v.is_a?(Time) || v.is_a?(Date) || v.is_a?(BigDecimal)}
           row.select {|k, v| v.to_s.strip.match(/^SELECT/i) }.each do |k, v|
             row[k] = [] unless row[k].is_a?(Array)
-            prepared_con.query(v.gsub(/\$\{([^\}]+)\}/, row[$1].to_s)).each do |nest_row|
```

Comment:
If database server is down, Here comes error.
This not Enclose begin and rescue.